### PR TITLE
proxychains-ng: init at 4.14

### PR DESCRIPTION
**Pull request by corps-fini**
###### Motivation for this change

Add proxychains-ng which is more maintained than proxychains.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/proxychains-ng/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  version = "4.14";
+
+  src = proxychains-ng-src-4-14;
+  doCheck = true;
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/rofl0r/proxychains-ng;
+    changelog = "https://github.com/rofl0r/proxychains-ng/tree/v${version}";
+    license = licenses.gpl2;
+    platforms = with platforms; linux ++ freebsd ++ netbsd ++ openbsd ++ darwin;
```
Comment (adisbladis) [inline pkgs/tools/networking/proxychains-ng/default.nix]: `platforms.unix` would be more appropriate.

File: pkgs/tools/networking/proxychains-ng/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+let
+  proxychains-ng-src-4-14 = fetchFromGitHub {
+    owner = "rofl0r";
+    repo = "proxychains-ng";
+    rev = "e895fb713ac1c1e92ee0795645f220573181c6fb";
+    sha256 = "03wk2xpxwc7kwlq6z9jf9df1vlh6p0dn0kf826fh1k7nfaa8c4py";
+  };
+in
+stdenv.mkDerivation rec {
+  name = "proxychains-ng-${version}";
```
Comment (adisbladis) [inline pkgs/tools/networking/proxychains-ng/default.nix]:
```suggestion
  pname = "proxychains-ng";
```

Then you don't even need to use `rec`.
`name` is derived from `pname` + `version`.

File: pkgs/tools/networking/proxychains-ng/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  version = "4.14";
+
+  src = proxychains-ng-src-4-14;
```
Comment (adisbladis) [inline pkgs/tools/networking/proxychains-ng/default.nix]:
```suggestion
  src = fetchFromGitHub {
    owner = "rofl0r";
    repo = "proxychains-ng";
    rev = "e895fb713ac1c1e92ee0795645f220573181c6fb";
    sha256 = "03wk2xpxwc7kwlq6z9jf9df1vlh6p0dn0kf826fh1k7nfaa8c4py";
  };
```

Looks much more idomatic.

**Review by adisbladis — CHANGES_REQUESTED** [review]

File: pkgs/tools/networking/proxychains-ng/default.nix (hunk 1 above)
Comment (godelized) [reply pkgs/tools/networking/proxychains-ng/default.nix]: Makes sense, thanks.

File: pkgs/tools/networking/proxychains-ng/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  pname = "proxychains-ng";
+  version = "4.14";
+
+  src = fetchFromGitHub {
+    owner = "rofl0r";
+    repo = "proxychains-ng";
+    rev = "e895fb713ac1c1e92ee0795645f220573181c6fb";
+    sha256 = "03wk2xpxwc7kwlq6z9jf9df1vlh6p0dn0kf826fh1k7nfaa8c4py";
+  };
+  doCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/proxychains-ng/default.nix:13]:
```suggestion
  };

  doCheck = true;
```

File: pkgs/tools/networking/proxychains-ng/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation {
+  pname = "proxychains-ng";
+  version = "4.14";
+
+  src = fetchFromGitHub {
+    owner = "rofl0r";
+    repo = "proxychains-ng";
+    rev = "e895fb713ac1c1e92ee0795645f220573181c6fb";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/proxychains-ng/default.nix:10]: Please track a tag.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please add it to top level.

**Comment by bobby285271** [comment]
Version 4.15 already in Nixpkgs.

**Pull request closed without merging**